tool/src/lib.rs

```diff
@@ -1,13 +1,225 @@
 // Enable once https://github.com/rust-lang/rust/issues/89554 is stable
 // #![deny(non_exhaustive_omitted_patterns)] // diplomat_core uses non_exhaustive a lot; we should never miss its patterns
 
+#[doc(hidden)]
 pub mod c;
+#[doc(hidden)]
 pub mod c2;
+#[doc(hidden)]
 pub mod common;
+#[doc(hidden)]
 pub mod cpp;
+#[doc(hidden)]
 pub mod cpp2;
-mod docs_util;
+#[doc(hidden)]
 pub mod dotnet;
+#[doc(hidden)]
 pub mod js;
+
+mod docs_util;
 mod layout;
 mod util;
+
+use colored::*;
+use core::panic;
+use diplomat_core::{ast, hir};
+use std::collections::HashMap;
+use std::fs::File;
+use std::io::Write;
+use std::path::Path;
+
+pub use ast::DocsUrlGenerator;
+
+pub fn gen(
+    entry: &Path,
+    target_language: &str,
```

Comment:
at the very least I think we should have a config struct or something that encapsulates all this

Reply:
we should also perhaps have a CLI feature